src/components/Tooltip/TooltipRenderedOnPageBody.js

```diff
@@ -0,0 +1,55 @@
+import React, {memo} from 'react';
+import PropTypes from 'prop-types';
+import {Animated, Text, View} from 'react-native';
+import ReactDOM from 'react-dom';
+
+const propTypes = {
+    // Style for Animation
+    // eslint-disable-next-line react/forbid-prop-types
+    animationStyle: PropTypes.object.isRequired,
+
+    // Syle for Tooltip wrapper
+    // eslint-disable-next-line react/forbid-prop-types
+    tooltipWrapperStyle: PropTypes.object.isRequired,
+
+    // Style for the text rendered inside tooltip
+    // eslint-disable-next-line react/forbid-prop-types
+    tooltipTextStyle: PropTypes.object.isRequired,
+
+    // Style for the Tooltip pointer Wrapper
+    // eslint-disable-next-line react/forbid-prop-types
+    pointerWrapperStyle: PropTypes.object.isRequired,
+
+    // Style for the Tooltip pointer
+    // eslint-disable-next-line react/forbid-prop-types
+    pointerStyle: PropTypes.object.isRequired,
+
+    // Callback to set the Ref to the Tooltip
+    setTooltipRef: PropTypes.func.isRequired,
+
+    // Text to be shown in the tooltip
+    text: PropTypes.string.isRequired,
+
+    // Callback to be used to calulate the width and height of tooltip
+    measureTooltip: PropTypes.func.isRequired,
+};
+
+const defaultProps = {};
+
+const TooltipRenderedOnPageBody = props => ReactDOM.createPortal(
+    <Animated.View
+        ref={props.setTooltipRef}
+        onLayout={props.measureTooltip}
+        style={[props.tooltipWrapperStyle, props.animationStyle]}
+    >
+        <Text style={props.tooltipTextStyle} numberOfLines={1}>{props.text}</Text>
+        <View style={props.pointerWrapperStyle}>
+            <View style={props.pointerStyle} />
+        </View>
+    </Animated.View>,
+    document.querySelector('body'),
+);
+
+TooltipRenderedOnPageBody.propTypes = propTypes;
+TooltipRenderedOnPageBody.defaultProps = defaultProps;
+export default memo(TooltipRenderedOnPageBody);
```

Comment:
Only use `memo()` where there is a measurable performance gain for doing so. In this case, I don't think there is any reason to use `memo()` unless there is something I'm missing?. More reading about this here: https://dmitripavlutin.com/use-react-memo-wisely/

Reply:
The props will change frequently. On every showtooltip  we update the position in state which will result in rerendering thus this component will rerender a lot and we update the state onLayout changes which will be triggered often. There will be n number of tooltip components in the page. Consider one for each group message Thus , I think its good to memorize this one.

Reply:
OK, thanks. I think that makes a pretty good case for it too. Would you mind adding that context as a code comment? That way the next person that sees this will know there was a specific reason it was added (as opposed to just adding it by default for no reason).